### DISCUSSION TITLE
Nextjs instead of CRA for SPA-like deployment

### DISCUSSION THREAD

**Discussion by CaptainN** [vercel/next.js#18591] [Help]
I have a project, which cannot leverage SSR or static caching, due to server constraints. We normally use CRA for that, but I'd like to use Nextjs instead. Primarily to use the router, which the team knows and loves, but also to gain zero-config route based code splitting, and the general consistency of tooling config across projects.

So the plan is to simply not define any getStaticProps or getServerSideProps methods, and use Nextjs just for spa like output - build then export. To get the routes aligned with the CMS, we'll use PHP to detect the incoming link structure, and simply serve up the right `[slug].js` route. For example, if we have a pathname like `/page/this-is-my-slug`, we'll explode the pathname, figure out that it's the `page` component, and load `/page/[slug].html` from the `out` directory. (I don't think it's strictly necessary to even do this - as I understand it we can just load the main index.html file, and that will work too - but this seems like a more next-like way to do it.)

This seems to work! The one issue is that it loads for a split second with the `router.asPath` prop set to `/page/[slug].html` instead of `/page/this-is-my-slug` (with `const router = useRouter()`). But it resolves immediately, and is easy to work around.

The point of this post is to ask whether I'm in for other unforeseen consequences of doing things this way. Am I?

**Comment by IskanderMustafin** [comment]
Since you don't use any data fetching methods I assume all of your pages are automatically statically optimised. And according to [documentation](https://nextjs.org/docs/routing/dynamic-routes#caveats)  there is some caveats when mixing both statically optimised pages and dynamic routes:  

 > Pages that are statically optimized by Automatic Static Optimization will be hydrated without their route parameters provided, i.e query will be an empty object ({}).

>After hydration, Next.js will trigger an update to your application to provide the route parameters in the query object.
